### ISSUE TITLE
ProjectionManager over HTTPS

### ISSUE THREAD

**Issue by aleksandrov** [EventStore/EventStore#1222]
Our current setup allows to communicate with ES REST API over HTTPS, it more less working, but that is not topic of the issue. 

I'm trying to use ProjectionsManager in the following way to connect using HTTPS
```csharp
[Test]
public async Task Test()
{
    var ip = await Dns.GetHostEntryAsync("es.host.name");
    var httpsEndpoint = new IPEndPoint(ip.AddressList.First(), 443);

    ProjectionsManager manager = new ProjectionsManager(Substitute.For<ILogger>(), httpsEndpoint, TimeSpan.FromMinutes(1));

    try
    {
        await manager.CreateTransientAsync("test", "query", new UserCredentials("admin", "changeit"));
    }
    catch (Exception ex)
    {
        // Server returned 400 (Bad Request) for POST on http://<resolved_ip>:443/projections/transient?name=test&type=JS    
        throw;
    }
}
```

ProjectionsManager still uses "http" instead of "https" because of [IPEndpointExtensions.cs](https://github.com/EventStore/EventStore/blob/master/src/EventStore.ClientAPI/Transport.Http/IPEndpointExtensions.cs) have "http" hard-coded.

Part of IPEndpointExtensions.cs
```csharp
        public static string ToHttpUrl(this IPEndPoint endPoint, string rawUrl = null)
        {
            return string.Format("http://{0}:{1}/{2}",
                                 endPoint.Address,
                                 endPoint.Port,
                                 rawUrl != null ? rawUrl.TrimStart('/') : string.Empty);
        }

        public static string ToHttpUrl(this IPEndPoint endPoint, string formatString, params object[] args)
        {
            return string.Format("http://{0}:{1}/{2}",
                                 endPoint.Address,
                                 endPoint.Port,
                                 string.Format(formatString.TrimStart('/'), args));
        }
```

It will be good to have ability to pass Uri to ProjectionsManager .ctor instead of IPEndPoint. In that case it will be much easier to detect correct protocol in IPEndpointExtensions class

**Comment by pgermishuys** [comment]
Thanks for opening this issue @aleksandrov. This is a known issue and will be resolved as part of this issue #1017

**Comment by aleksandrov** [comment]
@pgermishuys thanks for really quick response! I saw #1017 but did not realized that it is related.
